### ISSUE TITLE
.htaccess output causes 500 error

### ISSUE THREAD

**Issue by dansundy** [johngodley/redirection#520]
### Summary

`.htaccess` output is broken when saving.

### Actual Behavior

When I add target URLs with a variable such as `$1` the output in `.htaccess` repeats where the dollar sign is.

### Steps to reproduce

Create a redirection in an Apache redirect group with the following parameters and save:

* Source: `/blog/(.*)`
* Regex: `true`
* Target: `/$1`

The output I get looks like this:

```
# Created by Redirection
# Thu, 09 Nov 2017 18:44:54 +0000
# Redirection 2.9.1 - http://urbangiraffe.com/plugins/redirection/

<IfModule mod_rewrite.c>
RewriteRule blog/(.*) /
# Thu, 09 Nov 2017 18:44:54 +0000
# Redirection 2.9.1 - http://urbangiraffe.com/plugins/redirection/

<IfModule mod_rewrite.c>
RewriteRule blog/(.*) /$1 [R=301,L]
</IfModule>

 [R=301,L]
</IfModule>

# End of Redirection
```

If I hit save again I get an error within the WordPress dashboard with the following information:

```
Plugin: 2.9.1
WordPress: 4.8.3
PHP: 5.5.10
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36

Error: Unexpected token < in JSON at position 0 (undefined)
Action: red_set_redirect
Params: {"orderBy":"id","filterBy":"","filter":"","id":4,"url":"/blog/(.*)","title":"Test","regex":true,"match_type":"url","action_type":"url","position":2,"group_id":3,"action_code":301,"action_data":{"url":"/$1"}}
Code: 500 Internal Server Error
Raw: <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator,
 you@example.com and inform them of the time the error occurred,
and anything you might have done that may have
caused the error.</p>
<p>More information about this error may be available
in the server error log.</p>
</body></html>
```

### Environment

* Browser: Chrome
* Browser Version: 61.0.3163.100 (Official Build) (64-bit)
* WordPress Version: 4.8.3
* Redirection Version: 2.9.1

**Comment by johngodley** [comment]
Interesting! I was able to reproduce this as well. I'm preparing a new release so I'll try and get a fix for this included in that.

Thanks for the report.

**Comment by dansundy** [comment]
Hate to be a nag, but now the `$1` variable is just removed in the `.htaccess` file.

**Comment by johngodley** [comment]
Hmm, in my excitement at fixing one problem it seems I forgot to fix the other. Noted and will be fixed in the next release. If you want to fix it yourself right now then you can modify line 202 of `models/htaccess.php` and change:

	$text = preg_replace( self::INSERT_REGEX, $text, $existing );

To

	$text = preg_replace( self::INSERT_REGEX, str_replace( '$', '\\$', $text ), $existing );

**Comment by dansundy** [comment]
Thanks! I'll do that @johngodley.

**Comment by johngodley** [comment]
Fixed in 873ed6f551eee622729742b1ea39b8cb41468480
